File: Project Euler/Problem 2 - Even Fibonacci numbers/Answers/Answer.py
def fib(limit):
    a, b = 1, 2
    while a < limit:
        a, b = b, a+b
        yield a

upper_bound = int(input('Enter the value of the upper bound: '))
fib_list = [i for i in fib(upper_bound) if i%2 == 0]
print(sum(fib_list))

File: Project Euler/Problem 2 - Even Fibonacci numbers/Answers/Answer.swift
struct fib {
    private var value = 0
    private var (a, b) = (1, 2)

    init (limit: Int) {
        self.value = limit
    }

    mutating func getNext() -> Int? {
        while (a < value) {
            (a, b) = (b, b+a)
            return a
        }
        return nil
    }

    mutating func current() -> Int {
        return a
    }
}

print("Enter a number: ", terminator: "")
let limit = Int(readLine()!)
var summation: Int = 0
var value = fib(limit: limit!)
while (value.getNext() != nil) {
    if (value.current() % 2 == 0) {
        summation = summation + value.current()
    }
}
print(summation)